src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpInitializeStaticFieldsInline.cs

```diff
@@ -18,7 +18,7 @@ protected override bool InitialiesStaticField(SyntaxNode node, SemanticModel sem
         {
             var assignmentNode = (AssignmentExpressionSyntax)node;
 
-            return assignmentNode.FirstAncestorOrSelf<AssignmentExpressionSyntax>(x => x.IsKind(SyntaxKind.AddAssignmentExpression)) == null &&
+            return assignmentNode.FirstAncestorOrSelf<LambdaExpressionSyntax>() == null &&
```

Comment:
@mavasani Do we have a helper method for "is inside lambda or anonymous function"?

Reply:
I think we have an IOperation based helper for it, but not the syntax one. I know we have such a helper in Roslyn repo though.

Reply:
Are you thinking about this? https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs#L135

Reply:
Actually might be this one https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs#L17

Reply:
Let me know which helper you were thinking about so that I can add it to this PR and get it in.

Reply:
Ping @mavasani @sharwell

Reply:
Ideally, we would just convert this analyzer to be based on IOperation, and that would avoid language specific checks and also improve its performance. We can then also use our existing helpers to check if an operation has an ancestor operation with `OperationKind.AnonymousFunction`. However, that change would probably be outside the scope of this PR. @Evangelink - can you file a separate issue to convert this to an operation analyzer?

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/3544